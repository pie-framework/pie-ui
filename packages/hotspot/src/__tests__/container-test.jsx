import React from 'react';
import _ from 'lodash';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Konva from 'konva';

import Container from '../hotspot/container';
import HotspotComponent from '../hotspot/index';

Konva.isBrowser = false;

describe('HotspotComponent', () => {
  describe('renders', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = props => shallow(<HotspotComponent { ...props } />);
    });

    it('snapshot', () => {
      let w = wrapper();

      expect(w).toMatchSnapshot();
    });

    it('snapshot with rationale', () => {
      let w = wrapper({ rationale: 'This is rationale' });

      expect(w).toMatchSnapshot();
    });
  });
});

describe('Container', () => {
  let onSelectChoice, wrapper;

  const mkWrapper = (opts = {}) => {
    opts = _.extend(
      {
        classes: {
          base: 'base'
        },
        dimensions: {
          height: 0,
          width: 0
        },
        disabled: false,
        hotspotColor: 'rgba(137, 183, 244, 0.65)',
        imageUrl: '',
        isEvaluateMode: false,
        outlineColor: 'blue',
        session: { answers: [] },
        shapes: {
          rectangles: [],
          polygons: []
        }
      },
      opts
    );

    return shallow(<Container {...opts} onSelectChoice={onSelectChoice} />);
  };

  beforeEach(() => {
    onSelectChoice = jest.fn();
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    describe('outline color', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ outlineColor: 'red' });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe('hotspot color', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ hotspotColor: 'rgba(217, 30, 24, 0.65)' });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe('image', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ imageUrl: 'https://picsum.photos/id/102/200/300' });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe('shapes', () => {
      it('renders', () => {
        const wrapper = mkWrapper({
          shapes: {
            rectangles: [
              { id: '1', x: 5, y: 5, width: 5, height: 5 },
              { id: '2', x: 25, y: 25, width: 5, height: 5 }
            ],
            polygons: [
              { id: '3', points: [{ x: 94, y: 4 }, { x: 89, y: 4 }, { x: 36, y: 40 }]}
            ]
          },
          imageUrl: 'https://picsum.photos/id/102/200/300',
          dimensions: {
            width: 200,
            height: 300
          }
        });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
