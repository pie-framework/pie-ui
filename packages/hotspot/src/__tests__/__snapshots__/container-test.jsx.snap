// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container snapshots hotspot color renders 1`] = `
<Container
  classes={
    Object {
      "base": "Container-base-2 base",
      "image": "Container-image-3",
      "imageContainer": "Container-imageContainer-4",
      "resize": "Container-resize-6",
      "stage": "Container-stage-5",
    }
  }
  dimensions={
    Object {
      "height": 0,
      "width": 0,
    }
  }
  disabled={false}
  hotspotColor="rgba(217, 30, 24, 0.65)"
  imageUrl=""
  isEvaluateMode={false}
  onSelectChoice={[MockFunction]}
  outlineColor="blue"
  session={
    Object {
      "answers": Array [],
    }
  }
  shapes={
    Object {
      "polygons": Array [],
      "rectangles": Array [],
    }
  }
/>
`;

exports[`Container snapshots image renders 1`] = `
<Container
  classes={
    Object {
      "base": "Container-base-2 base",
      "image": "Container-image-3",
      "imageContainer": "Container-imageContainer-4",
      "resize": "Container-resize-6",
      "stage": "Container-stage-5",
    }
  }
  dimensions={
    Object {
      "height": 0,
      "width": 0,
    }
  }
  disabled={false}
  hotspotColor="rgba(137, 183, 244, 0.65)"
  imageUrl="https://picsum.photos/id/102/200/300"
  isEvaluateMode={false}
  onSelectChoice={[MockFunction]}
  outlineColor="blue"
  session={
    Object {
      "answers": Array [],
    }
  }
  shapes={
    Object {
      "polygons": Array [],
      "rectangles": Array [],
    }
  }
/>
`;

exports[`Container snapshots outline color renders 1`] = `
<Container
  classes={
    Object {
      "base": "Container-base-2 base",
      "image": "Container-image-3",
      "imageContainer": "Container-imageContainer-4",
      "resize": "Container-resize-6",
      "stage": "Container-stage-5",
    }
  }
  dimensions={
    Object {
      "height": 0,
      "width": 0,
    }
  }
  disabled={false}
  hotspotColor="rgba(137, 183, 244, 0.65)"
  imageUrl=""
  isEvaluateMode={false}
  onSelectChoice={[MockFunction]}
  outlineColor="red"
  session={
    Object {
      "answers": Array [],
    }
  }
  shapes={
    Object {
      "polygons": Array [],
      "rectangles": Array [],
    }
  }
/>
`;

exports[`Container snapshots shapes renders 1`] = `
<Container
  classes={
    Object {
      "base": "Container-base-2 base",
      "image": "Container-image-3",
      "imageContainer": "Container-imageContainer-4",
      "resize": "Container-resize-6",
      "stage": "Container-stage-5",
    }
  }
  dimensions={
    Object {
      "height": 300,
      "width": 200,
    }
  }
  disabled={false}
  hotspotColor="rgba(137, 183, 244, 0.65)"
  imageUrl="https://picsum.photos/id/102/200/300"
  isEvaluateMode={false}
  onSelectChoice={[MockFunction]}
  outlineColor="blue"
  session={
    Object {
      "answers": Array [],
    }
  }
  shapes={
    Object {
      "polygons": Array [
        Object {
          "id": "3",
          "points": Array [
            Object {
              "x": 94,
              "y": 4,
            },
            Object {
              "x": 89,
              "y": 4,
            },
            Object {
              "x": 36,
              "y": 40,
            },
          ],
        },
      ],
      "rectangles": Array [
        Object {
          "height": 5,
          "id": "1",
          "width": 5,
          "x": 5,
          "y": 5,
        },
        Object {
          "height": 5,
          "id": "2",
          "width": 5,
          "x": 25,
          "y": 25,
        },
      ],
    }
  }
/>
`;

exports[`HotspotComponent renders snapshot 1`] = `
<HotspotComponent
  classes={
    Object {
      "collapsible": "HotspotComponent-collapsible-1",
    }
  }
/>
`;

exports[`HotspotComponent renders snapshot with rationale 1`] = `
<HotspotComponent
  classes={
    Object {
      "collapsible": "HotspotComponent-collapsible-1",
    }
  }
  rationale="This is rationale"
/>
`;
